### PR TITLE
Allowing to reverse icon in DateInput

### PR DESCRIPTION
**Pull request by joaoGabriel55**
#### What does this PR do?

Fixed issue https://github.com/grommet/grommet/issues/6153 which consists of allowing to reverse icon in DateInput.

#### Where should the reviewer start?

Just access:

http://localhost:9001/?path=/story/input-dateinput-format--format

#### What testing has been done on this PR?

Unit test to Match Snapshot.

#### How should this be manually tested?

It is not necessary

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?

No

#### What are the relevant issues?

No

#### Screenshots

image

#### Do the grommet docs need to be updated?

Yes, to add the new DateInput prop, `reverse`

#### Should this PR be mentioned in the release notes?

No

#### Is this change backwards compatible or is it a breaking change?

No

### CODE REVIEW

**Review by jcfilben — COMMENTED** [review]
Just a few notes:
- We should support both passing the reverse prop through `inputProps` and having the `reverse` prop on DateInput. Ideally just using the `reverse` prop is preferred but we need to support passing reverse through `inputProps` for backwards compatibility since we supported this in the past. See https://github.com/grommet/grommet/pull/6160/files for a similar discussion about the icon prop on DateInput.

- When the icon is on the left side of DateInput the tab order is messed up. The focus moves first to the input and then the icon. It should move to the icon first and then the input since that is how it is visually ordered.

**Review by amandadupell — APPROVED** [review]
looks good to me! tab order goes from icon to input, following logical order. allows for `reverse` prop and passing in via `inputProps` for backwards compatibility

File: src/js/components/DateInput/stories/Format.js [hunk 1]
```diff
@@ -12,9 +13,30 @@ export const Format = () => {
...
       </Box>
+      <Box width="medium" margin={{ vertical: 'small' }}>
+        <DateInput format="m/d/yy" value={value} reverse onChange={onChange} />
+      </Box>
+      <Box width="medium" margin={{ vertical: 'small' }}>
+        <DateInput
+          format="m/d/yy"
+          value={value}
+          icon={<ClockIcon />}
+          inputProps={{ reverse: true }}
```
Comment (jcfilben) [inline src/js/components/DateInput/stories/Format.js:27]: I think we should encourage using the `reverse` prop over setting reverse through input props. Can we change the story examples to use `reverse`?

File: src/js/components/DateInput/DateInput.js [hunk 2]
```diff
@@ -111,6 +112,13 @@ Use the icon prop instead.`,
       );
     }
 
+    const reverse = reverseProp || restOfInputProps.reverse;
+
+    const calendarDropdownAlign = {
+      top: 'bottom',
+      ...(reverse ? { right: 'right' } : { left: 'left' }),
+    };
```
Comment (jcfilben) [inline src/js/components/DateInput/DateInput.js:120]: I think the alignment of the calendar should stay the same even if we use the reverse prop. The reverse prop should only affect the icon.

**Review by jcfilben — COMMENTED** [review]
Coming along nicely, just a few comments

File: src/js/components/DateInput/stories/Format.js (hunk 1 above)
Comment (joaoGabriel55) [reply src/js/components/DateInput/stories/Format.js:27]: Yes, that makes sense

File: src/js/components/DateInput/DateInput.js (hunk 2 above)
Comment (joaoGabriel55) [reply src/js/components/DateInput/DateInput.js:120]: If I don't do this the Input will be this way:

image

Would that be the expected behavior?

Comment (joaoGabriel55) [reply src/js/components/DateInput/DateInput.js:120]: Very nice tip

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**